web/src/inbox_ui.js

```diff
@@ -0,0 +1,875 @@
+    // being broken sometimes when we narrow
+    // to a filter and back to recent topics
+    // before it completely re-rerenders.
+    message_view_header.render_title_area();
+
+    // Fire our custom event
+    $("#message_feed_container").trigger("message_feed_shown");
+
+    // This makes sure user lands on the selected message
+    // and not always at the top of the narrow.
+    navigate.plan_scroll_to_selected();
+}
+
+function get_topic_key(stream_id, topic) {
+    return stream_id + ":" + topic;
+}
+
+function get_stream_key(stream_id) {
+    return "stream_" + stream_id;
+}
+
+function get_stream_container(stream_key) {
+    return $(`#${CSS.escape(stream_key)}`);
+}
+
+function get_topics_container(stream_id) {
+    const $topics_container = get_stream_header_row(stream_id).next();
+    return $topics_container;
+}
+
+function get_stream_header_row(stream_id) {
+    const $stream_header_row = $(`#${CSS.escape(STREAM_HEADER_PREFIX + stream_id)}`);
+    return $stream_header_row;
+}
+
+function load_filter() {
+    include_muted = ls.get(ls_key);
+    update_filters();
+}
+
+function update_filters() {
+    const $mute_checkbox = $("#inbox_filters #inbox_filter_mute_toggle");
+    const $mute_filter = $("#inbox_filters .btn-inbox-filter");
+    if (include_muted) {
+        $mute_checkbox.removeClass("fa-square-o");
+        $mute_checkbox.addClass("fa-check-square-o");
+        $mute_filter.addClass("btn-inbox-selected");
+    } else {
+        $mute_checkbox.removeClass("fa-check-square-o");
+        $mute_checkbox.addClass("fa-square-o");
+        $mute_filter.removeClass("btn-inbox-selected");
+    }
+}
+
+export function toggle_filter() {
+    const $mute_filter = $("#inbox_filters .btn-inbox-filter");
+    if ($mute_filter.hasClass("btn-inbox-selected")) {
+        include_muted = false;
+    } else {
+        include_muted = true;
+    }
+
+    update_filters();
+    save_filter();
+    update_data_and_ui();
+}
+
+function format_pm(user_ids_string) {
+    const recipient_ids = people.user_ids_string_to_ids_array(user_ids_string);
+    if (!recipient_ids.length) {
+        // Self DM
+        recipient_ids.push(people.my_current_user_id());
+    }
+
+    const unread_count = unread.num_unread_for_user_ids_string(user_ids_string);
+    const reply_to = people.user_ids_string_to_emails_string(user_ids_string);
+    const recipients_info = [];
+
+    for (const user_id of recipient_ids) {
+        const recipient_user_obj = people.get_by_user_id(user_id);
+        const user_circle_class = buddy_data.get_user_circle_class(user_id);
+        recipient_user_obj.user_circle_class = user_circle_class;
+
+        if (recipient_user_obj.is_bot) {
+            // Bots do not have status emoji, and are modeled as
+            // always present. We may want to use this space for a
+            // bot icon in the future.
+            recipient_user_obj.user_circle_class = "user_circle_green";
+        }
```

Comment:
I think we changed how we handle this for bots in "recent topics", and should probably adapt that logic here as well:
```
            if (user.is_bot) {                                                                          
                // We display the bot icon rather than a user circle for bots.                          
                context.is_bot = true;                                                                  
            } else {                                                                                    
                context.user_circle_class = buddy_data.get_user_circle_class(user_id);                  
            }                                                                                           
```

Reply:
Done

Reply:
How was this resolved? I don't see the same pattern as we have in the recent view, and I think this ends up setting an (unused) `user_circle_class` value for bots?  I'm not sure the intent but if we're doing that, we should at least document with a comment what we're doing.

Reply:
Oh yeah! Modified the logic like this:
```
        if (!recipient_user_obj.is_bot) {                                                                                                                                         
            const user_circle_class = buddy_data.get_user_circle_class(user_id);
            recipient_user_obj.user_circle_class = user_circle_class;
        }
```